languages/go/internal/host/host.go

```diff
@@ -69,14 +80,15 @@ func (h Host) getClass(name string) (*reflect.Type, error) {
 	return nil, errors.NewUnregisteredClassError(name)
 }
 
-func (h Host) CacheClass(cls reflect.Type, name string, constructor reflect.Value) error {
+func (h Host) CacheClass(cls reflect.Type, name string, constructor reflect.Value, fields map[string]interface{}) error {
 	if v, ok := h.classes[name]; ok {
 		return errors.NewDuplicateClassAliasError(name, cls, v)
 	}
 	h.classes[name] = cls
 	if constructor.IsValid() {
 		h.constructors[name] = constructor
 	}
+	h.fields[name] = fields
```

Comment:
This is different from how we do it now in python/ruby/js, which is to have one map where the values are type info structs. Since we no longer have 3 query callbacks per type it doesn't seem so bad to do it this way for now. I'll make a task for next cycle to bring it in line